ext/vl53/vl53_transport_impl.hpp

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2025, Henrik Hose
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_VL53_TRANSPORT_HPP
+#error "Don't include this file directly, use 'bmi088_transport.hpp' instead!"
+#endif
+
+namespace modm
+{
+
+template<typename SpiMaster, typename Cs, typename Lpn, size_t VL53LMZ_COMMS_CHUNK_SIZE>
+uint8_t
+Vl53SpiTransport<SpiMaster, Cs, Lpn, VL53LMZ_COMMS_CHUNK_SIZE>::writeMulti(uint16_t RegisterAddress,
+																		   const uint8_t *p_values,
+																		   uint32_t size)
+{
+	for (uint32_t position = 0; position < size; position += VL53LMZ_COMMS_CHUNK_SIZE)
+	{
+		uint32_t data_size = 0;
+		if (size > VL53LMZ_COMMS_CHUNK_SIZE)
+		{
+			if ((position + VL53LMZ_COMMS_CHUNK_SIZE) > size)
+			{
+				data_size = size - position;
+			} else
+			{
+				data_size = VL53LMZ_COMMS_CHUNK_SIZE;
+			}
+		} else
+		{
+			data_size = size;
+		}
+
+		const uint16_t temp = RegisterAddress + position;
+
+		addr_buffer[0] = SPI_WRITE_MASK(temp) >> 8;
+		addr_buffer[1] = SPI_WRITE_MASK(temp) & 0xFF;
+
+		modm::this_fiber::poll([&] { return this->acquireMaster(); });
+		Cs::reset();
+		// SpiMaster::transfer(&tx_buffer[0], nullptr, data_size);
+		SpiMaster::transfer(addr_buffer, nullptr, sizeof(addr_buffer));
+		SpiMaster::transfer(&p_values[position], nullptr, data_size);
```

Comment:
Dumb question: Why are we chunking this here? Is there a limit on the SPI transfer length? Cos the I2C just dumps 32kB of data into the sensor in one transfer, if I understand this correctly.

Reply:
The ST ULD implementation does this and due to the lack of documentation for the sensors besides the provided drivers, I just did the same.
I can test tomorrow if we can just write the entire firmware blob at once.